<CARRY>: Ocp docker file

kannon92 opened the pull request:
Commit 1 adds the ocp docker file
Commit 2 vendors the dependencies.

kannon92 commented:
/hold

merge after #26

ardaguclu commented on Dockerfile.ocp:
```diff
@@ -0,0 +1,27 @@
+# was called. For example, if we call make docker-build in a local env which has the Apple Silicon M1 SO
+# the docker BUILDPLATFORM arg will be linux/arm64 when for Apple x86 it will be linux/amd64. Therefore,
+# by leaving it empty we can ensure that the container and binary shipped on it will have the same platform.
+RUN CGO_ENABLED=${CGO_ENABLED:-1} GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} go build -a -o manager cmd/main.go
```
I'm wondering whether it is possible to use build-machinery-go like in https://github.com/openshift/cli-manager/blob/5c0184d6697a2d4e40f9610b763943549525fac8/Makefile#L14-L17 to manage all these things for us. But I understand that this is fork of upstream, if it is not possible.

kannon92 replied:
I'd rather not change this too much.

ardaguclu reviewed (commented):
Dropped my comments

kannon92 commented on Dockerfile.ocp:
```diff
@@ -0,0 +1,27 @@
+ARG TARGETARCH
+
+WORKDIR /workspace
+COPY . .
```
I don't see any failures on konflux at the moment.

kannon92 commented:
/retest

ardaguclu commented:
/retest

kannon92 commented:
/hold cancel

@ardaguclu 

Lets get this in now and we can iterate on konflux fixes.

ardaguclu commented:
I saw that main branch has already vendor directory. In that PR, there is no go.mod/sum changes. Is everything ok?

kannon92 commented:
> I saw that main branch has already vendor directory. In that PR, there is no go.mod/sum changes. Is everything ok?

Rebased. Should look better now.

I think I moved the vendor commit from this into a separate carry patch.

ardaguclu commented:
/lgtm
/hold
until Konflux passes

ardaguclu commented:
It looks like the Konflux failure is legit.

kannon92 commented:
@ardaguclu I fixed the konflux issues.

I had to vendor some stuff after rebase.

ardaguclu commented:
Great, thank you
/lgtm

kannon92 commented:
/hold cancel
